vignettes/UsingSelfControlledCohort.Rmd

```diff
@@ -0,0 +1,222 @@
+                             datatype = c("INTEGER",    "INTEGER",             "DATE",               "DATE"))
+
+outcome_table <- data.frame(field    = c("person_id", "outcome_concept_id", "outcome_start_date"),
+                            datatype = c("INTEGER",   "INTEGER",            "DATE"))
+
+knitr::kable(exposure_table)
+knitr::kable(outcome_table)
+```
+
+
+These tables can be created manually or standard CDM tables can be used (eg. drug_era => exposure_table, condition_era => outcome_table). Every exposure-outcome combination in these tables will be included in the analysis unless otherwise specified. The additional CDM tables that are required for this analyis are `observation_period` (for observation start and end dates) and `person` (for year of birth).
+
+# A first example
+
+The SelfControlledCohort package makes it easy to run examine the association between every drug and every condition in a CDM database. The Eunomia package provides a mini CDM that runs entirely within R that we can use for examples.
+
+
+```{r}
+# A first example with
+library(SelfControlledCohort)
+library(Eunomia)
+
+connectionDetails <- Eunomia::getEunomiaConnectionDetails()
+
+result <- runSelfControlledCohort(
+  connectionDetails = connectionDetails,
+  cdmDatabaseSchema = "main",
+  exposureIds = '', # inlcude all drugs as exposures
+  outcomeIds = '' # include all conditions as outcomes
+)
+
+library(dplyr)
+result$estimates %>% 
+  arrange(desc(irr)) %>% 
+  head()
+```
+
+The result of our analysis is a dataframe with one row per exposure-outcome pair along with all the relevant statistics. Let's interpret the resluts for amoxicillin exposure (concept ID 1713332) and the outcome of Chronic sinusitis (concept ID 257012)
+
+```{r}
+example <- result$estimates %>% 
+  filter(exposureId == 1713332, outcomeId == 257012) 
+
+example %>% 
+  tidyr::gather() %>% 
+  mutate(value = format(round(value,1), scientific = F)) %>% 
+  rename(column = key) %>% 
+  knitr::kable(align = c("lr"))
+```
+
+We can see that `r example$numPersons` were exposed to amoxicillin. When we add up all the time before the exposure we get `r example$timeAtRiskUnexposed` person-years. Similarly when we add up all the time after the exposure, the "Exposed time at risk", we get `r example$timeAtRiskExposed` person-years.
+
+The incidence of Chronic sinusitis during the "Unexposed time at risk" is
+
+$$
+\frac{224 \ \ events}{45796.0 \ \ person \ years} =  0.00489
+$$
+
+
+The incidence of Chronic sinusitis during the "Exposed time at risk" is
+
+
+$$
+\frac{65 \ \ events}{277.4 \ \ person \ years} =  0.234
+$$
+
+The incidence rate ratio is $ \frac{234}{0.00489} = 47.9$ with a 95% confidence interval of $[35.8, 63.4]$
+
+When the sample size is large the log of the relative risk is normally distributed with a mean of $logRr = 3.9$ and a standard error of $seLogRr =	0.1$ <Not sure this is correct actually>
```

Comment:
Need to check this.

Reply:
If you're referring to the line in the code to get the seLogRr:

`estimates$seLogRr <- (log(estimates$irrUb95) - log(estimates$irrLb95))/(2 * qnorm(0.975))`

This is the standard error of the log of the irr value please correct me if I'm wrong, but I think that the log estimates are used to compare the magnitude of the effect, not that the risk ratios are log-normally distributed.

Reply:
I think I'll leave this out and just reference the vignette. My main question was if the confidence intervals were valid for small sample sizes and I think they are.